### PR TITLE
Update to JavaScriptCore 4.1/Update to 2021 edition

### PR DESCRIPTION
**Pull request by mrquantumoff**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [x] Refactor
- [x] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [ ] No
- [x] I'm not sure, I just updated javascriptcoregtk from 4.0 to 4.1

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix #___, #___`)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [x] I have added a convincing reason for adding this feature

### Other information

In order to package an app for flatpak without sacrificing new security/other features  and user's hard disk space by using an older runtime this package would need an update to JavaScriptCore 4.1

### CODE REVIEW

**Comment by FabianLars** [comment]
Thanks for contributing!

To be honest, i am not a fan of the new folder, primarily because the old layout seems to be the default in similar crates as far as i can see.

Also we should probably release 4.0 with `0.16` dependencies first, and then this so that we can keep all 3 versions around (we need 4.0 + 0.15 for Tauri for example) -> This is more of a message to the other maintainers.

There may be a bit delay until Yu-Wei reviews it because we're quite busy right now and aren't completely sure yet what the best upgrade strategy would be.

**Comment by mrquantumoff** [comment]
> Thanks for contributing!
> 
> To be honest, i am not a fan of the new folder, primarily because the old layout seems to be the default in similar crates as far as i can see.
> 
> Also we should probably release 4.0 with `0.16` dependencies first, and then this so that we can keep all 3 versions around (we need 4.0 + 0.15 for Tauri for example) -> This is more of a message to the other maintainers.
> 
> There may be a bit delay until Yu-Wei reviews it because we're quite busy right now and aren't completely sure yet what the best upgrade strategy would be.

Ok, I'll reverse the whole new folder thing when I get the chance, I'll wait as much as I'll have to, I get that maintaining OSS isn't as easy as many people think it is

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
         if: matrix.platform.os == 'ubuntu-latest'
         run: |
           sudo apt-get update
-          sudo apt-get install -y webkit2gtk-4.0
+          sudo apt-get install -y webkit2gtk-4.1
```
Comment (wusyong) [inline .github/workflows/build.yml]: I think we can publish a minor version with 2021 edition and using glib 0.16.
But we might need to stick in webkit2gtk-4.0 as many old distribution might not have 4.1
We'll look into 4.1 after we switch to flatpak for packaging and CICD.

```suggestion
          sudo apt-get install -y webkit2gtk-4.0
```

File: sys/Cargo.toml [hunk 2]
```diff
@@ -1,39 +1,36 @@
...
+version = "0.4.0"
 
 [package.metadata.system-deps.javascriptcoregtk_4_0]
-name = "javascriptcoregtk-4.0"
-version = "2.24"
-
-  [package.metadata.system-deps.javascriptcoregtk_4_0.v2_28]
-  version = "2.28"
+name = "javascriptcoregtk-4.1"
+version = "2.38"
```
Comment (wusyong) [inline sys/Cargo.toml]:
```suggestion
name = "javascriptcoregtk-4.0"
version = "2.24"
```

File: sys/tests/abi.rs [hunk 3]
```diff
@@ -14,7 +14,7 @@ use std::{
 };
 use tempfile::Builder;
 
-static PACKAGES: &[&str] = &["javascriptcoregtk-4.0"];
+static PACKAGES: &[&str] = &["javascriptcoregtk-4.1"];
```
Comment (wusyong) [inline sys/tests/abi.rs]:
```suggestion
static PACKAGES: &[&str] = &["javascriptcoregtk-4.0"];
```

File: Gir.toml [hunk 4]
```diff
@@ -1,63 +1,63 @@
 [options]
+deprecate_by_min_version = true
+girs_directories = ["gir-files/"]
 library = "JavaScriptCore"
-version = "4.0"
+min_cfg_version = "2.38"
+single_version_file = true
 target_path = "."
-min_cfg_version = "2.24"
+version = "4.1"
```
Comment (wusyong) [inline Gir.toml]:
```suggestion
version = "4.0"
```

File: Gir.toml [hunk 5]
```diff
@@ -1,63 +1,63 @@
 [options]
+deprecate_by_min_version = true
+girs_directories = ["gir-files/"]
 library = "JavaScriptCore"
-version = "4.0"
+min_cfg_version = "2.38"
```
Comment (wusyong) [inline Gir.toml]:
```suggestion
min_cfg_version = "2.24"
```

**Review by wusyong — CHANGES_REQUESTED** [review]
Just need to revert a few version changes and I think it should pass the CI.

**Review by wusyong — APPROVED** [review]

**Pull request merged**